roles/servicemesh/tasks/main.yml

```diff
@@ -0,0 +1,343 @@
+---
+- name: Get current state
+  k8s_info:
+    api_version: ripsaw.cloudbulldozer.io/v1alpha1
+    kind: Benchmark
+    name: '{{ meta.name }}'
+    namespace: '{{ operator_namespace }}'
+  register: resource_state
+
+- name: Define variables
+  set_fact:
+    servicemesh: "{{ servicemesh_defaults | combine(workload_args, recursive=True) }}"
+    controlplane_namespace: "{{ meta.name }}-controlplane-{{ trunc_uuid }}"
+    workload_namespace: "{{ meta.name }}-workload-{{ trunc_uuid }}"
+    owner:
+      - apiVersion: ripsaw.cloudbulldozer.io/v1alpha1
+        kind: Benchmark
+        name: "{{ meta.name }}"
+        namespace: "{{ operator_namespace }}"
+        uid: "{{ resource_state.resources[0].metadata.uid }}"
+
+- when: resource_state.resources[0].status.state is not defined
+  block:
+  - name: Create control plane namespace
+    k8s:
+      definition:
+        apiVersion: project.openshift.io/v1
+        kind: ProjectRequest
+        metadata:
+          name: "{{ controlplane_namespace }}"
+
+  - name: Create ServiceMesh control plane
+    k8s:
+      definition: "{{ lookup('template', 'smcp.yaml.j2') | from_yaml }}"
+
+  - name: Create workload namespace
```

Comment:
Instead of having the operator create the namespace, can we just not have the prereq to have this namespace already created?

Reply:
My take on this is that what can be automated should be automated. ProjectRequest RBAC is not something too security sensitive, so why not give the right to the operator instead of having it as prereq?

Reply:
I'm ok w/ this given the user has to make the choice to allow the perms and apply the crd. So there is an implied responsibility on the user